tests/Suggestor/Suggestor.test.ts

```diff
@@ -150,3 +150,68 @@ describe.each([
         markdownTable.verify();
     });
 });
+
+describe('onlySuggestIfBracketOpen', () => {
+    const buildSuggestions: SuggestionBuilder = onlySuggestIfBracketOpen(
+        // A dummy SuggestionBuilder. Just used to validate that results are returned, results aren't meaningful
+        () => [{} as unknown as SuggestInfo],
+        [
+            ['(', ')'],
+            ['[', ']'],
+        ],
+    );
+
+    /**
+     * Given a string with **exactly** one vertical bar (`|`), returns the string with the vertical bar removed
+     * and the index of the bar.
+     *
+     * This is used as a helper when writing tests, since it provides a less error prone
+     * and more readable way to represent a cursor's position, denoted by the vertical bar, and the string.
+     *
+     * @param line - A string that contains exactly one vertical bar (`|`)
+     * @returns A tuple of the line without the vertical bar, and the index of the vertical bar.
+     */
+    function cursorPosition(line: string): [line_without_cursor: string, custor_index: number] {
+        const line_without_cursor = line.replace('|', '');
+        // Check that the cursor marker appears exactly once in each input string:
+        expect(line_without_cursor.length).toEqual(line.length - 1);
+        console.log([line.replace('|', ''), line.indexOf('|')]);
+        return [line.replace('|', ''), line.indexOf('|')];
+    }
+
+    it('should suggest if cursor at end of line with an open pair', () => {
+        for (const line of ['(hello world|', '[hello world|']) {
+            const suggestions = buildSuggestions(...cursorPosition(line), getSettings());
+            expect(suggestions).not.toEqual([]);
+        }
+    });
+
+    it('should suggest if cursor at end of line with an nested open pairs', () => {
+        for (const line of ['(((hello world))|', '[[[hello world]]|']) {
+            const suggestions = buildSuggestions(...cursorPosition(line), getSettings());
+            expect(suggestions).not.toEqual([]);
+        }
+    });
+
+    it('should suggest if cursor in middle of closed pair', () => {
+        for (const line of ['(hello world|)', '[hello world|]']) {
+            const suggestions = buildSuggestions(...cursorPosition(line), getSettings());
+            expect(suggestions).not.toEqual([]);
```

Comment:
I've read these tests multiple times and have to think about them each time.

I think it's worth extracting `[]` as:

```ts
    const emptySuggestion = [] as unknown;
```

So the tests become things like:

```diff
     it("should not suggest if there's no open bracket at cursor position", () => {
         const suggestions = buildSuggestions(...cursorPosition('(hello world)|'), getSettings());
-        expect(suggestions).toEqual([]);
+        expect(suggestions).toEqual(emptySuggestion);
     });
 });
```

It just saves having to work out what the specific meaning of `[]` is...